OpenRA.Game/Server/Server.cs

```diff
@@ -882,6 +1036,12 @@ public void DropClient(Connection toDrop)
 				// Send disconnected order, even if still in the lobby
 				DispatchOrdersToClients(toDrop, 0, Order.FromTargetString("Disconnected", "", true).Serialize());
 
+				if (gameInfo != null && !dropClient.IsObserver)
+				{
+					var disconnectedPlayer = gameInfo.Players.FirstOrDefault(p => p.ClientIndex == toDrop.PlayerIndex);
+					disconnectedPlayer.DisconnectFrame = toDrop.MostRecentFrame;
```

Comment:
Style nit: `.FirstOrDefault(...).anything` is a code smell because the `OrDefault` will trigger a null-reference exception. In this case we know that there will always be a match, so we can change this to just `.First(...)`. Alternatively we can keep the `.FirstOrDefault(...)` and wrap this in a null check.